### ISSUE TITLE
Sending `onAnimatedValueUpdate` with no listeners registered `warning` when swiping back using ios gesture. JS Stack Navigator 

### ISSUE THREAD

**Issue by bryanltobing** [react-navigation/react-navigation#11564]
### Current behavior

This is only happening on JS Stack Navigator 

I keep getting warning that says "Sending `onAnimatedValueUpdate` with no listeners registered" every time i'm swiping back to the previous screen using ios gesture. 

I'm not sure why since i'm following the installation and setup in the docs correctly. 

## Screenshot

![image](https://github.com/react-navigation/react-navigation/assets/46083126/547e449c-884f-45fb-93d8-0fc6e3cfc631)

## Video
https://github.com/react-navigation/react-navigation/assets/46083126/b69c46af-49d4-48e5-a605-c5aa9d79685e

### Expected behavior

There should not be warning for gesture back behavior. This is sometimes really annoying when debugging or developing my apps.

### Reproduction

https://github.com/bryanltobing/rnav-test

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [ ] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 6.1.7
| @react-navigation/stack                | 6.3.17
| react-native-safe-area-context         | 4.6.3
| react-native-screens                   | 3.22.0
| react-native-gesture-handler           | 2.12.0
| react-native                           | 0.72.4
| expo                                   | 49.0.8
| node                                   | 20.3.1
| npm or yarn                            | yarn 1.22.19

**Comment by cyberkaidev** [comment]
I have the same problem when I use:
```
    navigation.dispatch(
        CommonActions.reset({
            index: 1,
            routes: [{ name: 'TabsRoutes' }],
        }),
    );
```

**Comment by DDM16** [comment]
same problem on me. 

i get warn after upgrading react version to 0.72.

**Comment by watislaf** [comment]
same problem.

**Comment by matterwin** [comment]
Problem still occurring here

**Comment by hwangJi-dev** [comment]
same problem

**Comment by xaiamov** [comment]
This hack helped me: https://stackoverflow.com/a/76915974/4177749
```js
navigation.removeListener();
navigation.navigate(tab, { screen, params });
```
```json
"react-native": "0.72.3",
"@react-navigation/stack": "6.3.17"
```

**Comment by GuidoGagliardini** [comment]
same issue here
```

    "@react-navigation/drawer": "^6.6.5",
    "@react-navigation/native": "^6.1.9",
    "@react-navigation/stack": "^6.3.20",
    "expo": "~49.0.13",
    "react": "18.2.0",
```

**Comment by xaiamov** [comment]
[This simple patch](https://github.com/react-navigation/react-navigation/issues/7839#issuecomment-1621988247) helped me.


```json
     "react-native": "0.72.3",
     "@react-navigation/bottom-tabs": "^6.5.8",
     "@react-navigation/native": "^6.1.7",
     "@react-navigation/stack": "^6.3.17",
```

**Comment by GuidoGagliardini** [comment]
> [This simple patch](https://github.com/react-navigation/react-navigation/issues/7839#issuecomment-1621988247) helped me.
> 
> ```json
>      "react-native": "0.72.3",
>      "@react-navigation/bottom-tabs": "^6.5.8",
>      "@react-navigation/native": "^6.1.7",
>      "@react-navigation/stack": "^6.3.17",
> ```
> 
> Duplicate #7839

Yes, I have tested it and it works but I think it is correct to report that the error still exists. Thanks anyway and best regards!

**Comment by Ero-pixel** [comment]
<RestaurantStack.Screen name="Restaurants" component={RestaurantsScreen}  options={{ title: 'Restaurants' }}/>
просто добавьте   options={{ title: 'Restaurants' }}  и всё

**Comment by CTOverton** [comment]
> [This simple patch](https://github.com/react-navigation/react-navigation/issues/7839#issuecomment-1621988247) helped me.
> 
> ```json
>      "react-native": "0.72.3",
>      "@react-navigation/bottom-tabs": "^6.5.8",
>      "@react-navigation/native": "^6.1.7",
>      "@react-navigation/stack": "^6.3.17",
> ```
> 
> Duplicate #7839

Would love to see some movement on this ticket. While this patch does work it seems like something that should be updated in the repo itself. 

@Ero-pixel Adding the title for me did not resolve my issue sadly.

**Comment by MarlonAEC** [comment]
> ```
>   <RestaurantStack.Screen name="Restaurants" component={RestaurantsScreen}  options={{ title: 'Restaurants' }}/>
> ```
> 
> Just add options={{ title: 'Restaurants' }} and that's it.

This worked for me. Thanks!

**Comment by tbagskk** [comment]
> [Ce simple patch](https://github.com/react-navigation/react-navigation/issues/7839#issuecomment-1621988247)

Ca a fonctionné pour moi aussi  !

**Comment by ahmadkhalaf1** [comment]
no update on this? nothing from above is working , ios

**Comment by wanderhungerbuhler** [comment]
> ```
>   <RestaurantStack.Screen name="Restaurants" component={RestaurantsScreen}  options={{ title: 'Restaurants' }}/>
> ```
> 
> Just add options={{ title: 'Restaurants' }} and that's it.

Add options={{ title: ... }} resolved for me.

**Comment by MariuzM** [comment]
Same issue when using `const Tab = createMaterialTopTabNavigator();` and swiping left to edge of screen i get this warning

**Comment by chris-hud** [comment]
I get this warning with material top tabs, options={{ title: ... }} does not resolve for me

**Comment by therbta** [comment]
Same issue here.

<Stack.Screen
          name="Chat"
          component={Conversation}
          options={{
            headerShown: false,
            presentation: 'fullScreenModal'
          }}
        />

**Comment by esmailbenmoussa** [comment]
> > ```
> >   <RestaurantStack.Screen name="Restaurants" component={RestaurantsScreen}  options={{ title: 'Restaurants' }}/>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Just add options={{ title: 'Restaurants' }} and that's it.
> 
> Add options={{ title: ... }} resolved for me.

This did not work for me..

**Comment by FancyLobster** [comment]
Any update on this? None of the fixes above worked for me. Here are my dependencies:

    "react": "18.2.0",
    "react-native": "0.73.6",
    "@react-navigation/material-top-tabs": "^6.6.13",
    "@react-navigation/native": "^6.1.17",
    "@react-navigation/stack": "^6.3.29",
    "react-native-gesture-handler": "^2.15.0",
    "react-native-reanimated": "^3.8.1",

**Comment by rocalex** [comment]
same issue

**Comment by Danu-rxone** [comment]
solution
https://stackoverflow.com/a/76465300/12285105

**Comment by Danu-rxone** [comment]
You can find solution at the following link: 
https://stackoverflow.com/a/76465300/12285105
I placed the code in the App.tsx file.

**Comment by FancyLobster** [comment]
> You can find solution at the following link: https://stackoverflow.com/a/76465300/12285105 I placed the code in the App.tsx file.

Thank you for the solution, but this did not work for me. I tried placing a listener in various places in my index.js, App.js, and Home.js and I still get the warning after I swipe between pages. I even added listeners for all my Animated and Shared Values that I use throughout the app and it still did not fix it. I also tried the other solutions in that post and they did not work.

Has anyone solved this issue with a method that differs from the solutions proposed in this thread or the referenced StackOverflow post?

**Comment by bmg817** [comment]
Experiencing this as well when I use navigation within an app to go back to the main screen

**Comment by JeffGuKang** [comment]
> ```
>   <RestaurantStack.Screen name="Restaurants" component={RestaurantsScreen}  options={{ title: 'Restaurants' }}/>
> ```
> 
> Just add options={{ title: 'Restaurants' }} and that's it.

I fixed this issue with above solution. But `options` should be option.

**Comment by ReiDx** [comment]
any update?

same issue

**Comment by FancyLobster** [comment]
> any update?
> 
> same issue

Still no fix, **I am just ignoring the warning until it is fixed**.

This only prevents it from popping up over the UI, but still does **not** prevent it from showing in the console log:

```
// index.js
import { LogBox } from 'react-native';
LogBox.ignoreLogs(['Sending `onAnimatedValueUpdate` with no listeners registered.']);
```

Funny enough, I thought the issue somehow stemmed from our app's complex/nested navigation setup, but that turns out to **not** be the case. I have been debugging in a test build environment with only 1 navigator and 2 pages and I am still getting the warning when navigating between those 2 pages. Maybe they need to just comment out the warning in the source code? I don't have time to look through it, but it is still quite annoying for it to be showing up in my console log.

**Comment by colaquecez** [comment]
Same here using `@react-navigation/material-top-tabs`

**Comment by emilje** [comment]
Same.

```
    "@react-navigation/bottom-tabs": "^6.5.20",
    "@react-navigation/elements": "^1.3.30",
    "@react-navigation/native": "^6.1.17",
    "@react-navigation/native-stack": "^6.9.26",
    "@react-navigation/stack": "^6.3.29",
```

**Comment by hoangphucnst** [comment]
> same issue on the react-native 0.74.1 ![image](https://private-user-images.githubusercontent.com/30758309/330357228-a6826dbd-d50b-4bb6-945a-618cafd06698.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2NDQ3OTQsIm5iZiI6MTcxNzY0NDQ5NCwicGF0aCI6Ii8zMDc1ODMwOS8zMzAzNTcyMjgtYTY4MjZkYmQtZDUwYi00YmI2LTk0NWEtNjE4Y2FmZDA2Njk4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDAzMjgxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlkZWViYzhmNjczZjFiMWUxZjkyMzA0Y2ExNDcxNGE0OGIxZjc3YjczMjBkZWE5NzA0M2IzYjRhZWQ5MTYxNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.p-d47o7VJHlCuhmQkyhVEbILEEEIHjzPl4AdgiGHj_I)
> 
> ```
> "react-native": "0.74.1",
> "@react-navigation/native": "^6.1.17",
> "@react-navigation/native-stack": "^6.9.26",
> "@react-navigation/stack": "^6.3.29",
> ```
> 
> the options={{title: 'screen title'}} and this : `import { Animated } from "react-native";
> 
> const av = new Animated.Value(0); av.addListener(() => {return});`
> 
> Doesn't work to me

Me too

**Comment by Catnais88** [comment]
"react-native": "^0.74.1",
"@react-navigation/bottom-tabs": "^6.5.8",
"@react-navigation/native": "^6.1.7",
"@react-navigation/native-stack": "^6.9.13",
"@react-navigation/stack": "^6.3.17",

Same for me

**Comment by thomaskyr** [comment]
> Same issue when using `const Tab = createMaterialTopTabNavigator();` and swiping left to edge of screen i get this warning
> 
> <img alt="image" width="1054" src="https://private-user-images.githubusercontent.com/36937039/305197818-614a71a7-b44e-49d9-be4d-88a7c73259e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNjYzNzksIm5iZiI6MTcyMDE2NjA3OSwicGF0aCI6Ii8zNjkzNzAzOS8zMDUxOTc4MTgtNjE0YTcxYTctYjQ0ZS00OWQ5LWJlNGQtODhhN2M3MzI1OWU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA1VDA3NTQzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3MDc3OTMzMTU1ZWE2OGJkNTAyNzA3Mjg0ODI2MjA4ZTE3ZmJmMGIyOGE1OGZmNDdjNDBkMzY5YmQyY2UwZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VaJ1kNmfLeGK9KZ8dcmND0hYby2iC4my4AKCTgHlsUk">


I had exactly the same issue trying to create a custom tab bar.
In order to resolve it, I adjusted the solution @Danu-rxone suggested:

> You can find solution at the following link: https://stackoverflow.com/a/76465300/12285105 I placed the code in the App.tsx file.

For animation interpolating I'm using position that's provided by react-navigation
```
const Tab = createMaterialTopTabNavigator();

 <Tab.Navigator
      tabBar={props => (
        <CustomTabBar {...props} />
      )} 
[...]
>
```

```
const CustomTabBar = ({
  state,
  descriptors,
  navigation,
  position,
})
```

I'm using this position property for animation:

```
const inputRange = state.routes.map((_, i) => i);
        const opacity = position.interpolate({
          inputRange,
          outputRange: inputRange.map(i => (i === index ? 1 : 0.5)),
        });
```

So what I did was add a listener for the animation I'm actually using (and not create a new, unused animation):

```
position.addListener(() => {
    return;
  });
```

Hope that helps. Cheers

**Comment by rsereir** [comment]
Same issue for me

**Comment by BangDori** [comment]
Same issue for me.

<div align="center">
  <img src="https://github.com/user-attachments/assets/485dcda7-124f-4e30-b777-8a71473de1c8" width="480" />
</div>

**Comment by RoyRao2333** [comment]
In my case, use `GestureHandlerRootView` to wrap `NavigationContainer` resolved the warnings:

```jsx
import { GestureHandlerRootView } from 'react-native-gesture-handler';

<GestureHandlerRootView style={{ flex: 1 }}>
  <NavigationContainer>{/* other codes */}</NavigationContainer>
</GestureHandlerRootView>
```

**Comment by johannessachse** [comment]
@RoyRao2333 Which versions are you using? In my case wrapping the `NavigationContainer` with `GestureHandlerRootView` makes no difference.

```json
// package.json
"@react-navigation/bottom-tabs": "6.5.11",
"@react-navigation/drawer": "6.7.2",
"@react-navigation/native": "6.1.18",
"@react-navigation/stack": "6.4.1",
"react-native-gesture-handler": "2.17.1",

```

**Comment by RoyRao2333** [comment]
@johannessachse 

It worked for most scenarios, but the warning would appear in some cases, just not that frequent. I guess this is not the ultimate solution. Sorry.

**Comment by dengcqw** [comment]
> Same issue when using `const Tab = createMaterialTopTabNavigator();` and swiping left to edge of screen i get this warning
> 
> <img alt="image" width="1054" src="https://private-user-images.githubusercontent.com/36937039/305197818-614a71a7-b44e-49d9-be4d-88a7c73259e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2NTgwNDAsIm5iZiI6MTcyNDY1Nzc0MCwicGF0aCI6Ii8zNjkzNzAzOS8zMDUxOTc4MTgtNjE0YTcxYTctYjQ0ZS00OWQ5LWJlNGQtODhhN2M3MzI1OWU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI2VDA3MzU0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNGIzOGFhZTYyODQ4ZWI0Y2UxNDExZDllNmM3NWE3NTZhNWEwN2RkZWM3NDAxNGEyM2FmZmZhZDQ5MzUzOWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.in9QEOiqInW6JH69guYadgjEtwUA-5wFcnbnecU-dqo">

import * as React from 'react';
import { Animated } from 'react-native';

export function useAnimatedValue(initialValue: number) {
  const lazyRef = React.useRef<Animated.Value>();

  if (lazyRef.current === undefined) {
    lazyRef.current = new Animated.Value(initialValue);
    lazyRef.current.addListener((_) => {}); // add default listener here then create patch
  }

  return lazyRef.current as Animated.Value;
}

**Comment by mayfairr** [comment]
Is this fixed?

**Comment by emilje** [comment]
> Is this fixed?

Nope, but it does not really cause any issues except being annoying... (as far as I can tell)

**Comment by mayfairr** [comment]
> > Is this fixed?
> 
> Nope, but it does not really cause any issues except being annoying... (as far as I can tell)

Ok thanks, I will just ignore using Logbox for now 🙏🏽

**Comment by upp22** [comment]
Still get the bug with `@react-navigation/stack`. If you don't need the customizations offered by react-navigation/stack, consider the native-stack which offers native performance and doesn't surface the errors. I understand the error to be harmless, but I'd prefer to not ignore them (personal preference). 

In the code above the stack was created using `createStackNavigator` from `@react-navigation/stack`. 

Import and create the [Native Stack Navigator](https://reactnavigation.org/docs/native-stack-navigator/):

> import { createNativeStackNavigator } from "@react-navigation/native-stack";

> const MyStack = createNativeStackNavigator<MyStackParamList>();

**Comment by apapalillo** [comment]
Why is this even a warning?

**Comment by pilotpirxie** [comment]
One and half years since opening the issue and the problem still persists even with freshly installed libs after following the official guide

**Comment by relense** [comment]
I also have this issue, for me it happens when the user swipes up and the app becames inactive and goes to background.

**Comment by oakmegaeddie** [comment]
same here, 2024-12-19 🥰

**Comment by MariuzM** [comment]
omg still issue same issue :(

**Comment by dave-mp** [comment]
Same here, seems to be happening when `navigation.reset` is called

**Comment by guyeilon** [comment]
same :)

**Comment by dave-mp** [comment]
Still happening in the latest version. Seems to be triggered when calling `navigation.reset`.

**Comment by MarlonAEC** [comment]
At this point, I just gave up on this issue and ignored it in the logs, this is not a priority it seems it's not going to get fixed anytime soon, unfortunately :(

**Comment by semihyagli** [comment]
Just encountered this today. One of the comments above suggested this simple and beautiful patch by [iaroslav-ternovyi](https://github.com/iaroslav-ternovyi): 

https://github.com/react-navigation/react-navigation/issues/7839#issuecomment-1621988247

This resolved my problem, not sure why react-navigation did not include this in a PR. I see that there are still people suffering from this, thought re-mentioning it, since it is a bit obscured by the new comments. 

Just add the two lines mentioned in the above patch on the designated lines inside the react-navigation/stack and run 
`npx patch-package @react-navigation/stack`

**Comment by LukasMod** [comment]
I encountered this issue when calling navigation.navigate() rapidly inside a useEffect that runs after the screen transition. To resolve it, I wrapped the navigation call using InteractionManager to ensure it executes only after all interactions and transitions have completed. Here's the utility function I used:

```
export const postponeExecution = (functionToPostpone: () => void) => {
  InteractionManager.runAfterInteractions(() => {
    setTimeout(() => {
      functionToPostpone();
    }, 0);
  });
};
```
Using this wrapper resolved the problem by deferring the navigation until the UI was fully ready.

```
  useEffect(() => {

      postponeExecution(() => {
        navigation.navigate('ScreenB')
      })
    
  }, [])
```

**Comment by ALFmachine** [comment]
i found that [this stack overflow comment](https://stackoverflow.com/a/79401814/5125846) from [MohammadAkil Chaklawala](https://stackoverflow.com/users/22403485/mohammadakil-chaklawala) resolved the warning issue for me simply.

TLDR; add an `initialRouteName` to the Stack.
